src/store-api/src/metadata.rs

```diff
@@ -967,6 +996,21 @@ pub enum MetadataError {
         location: Location,
     },
 
+    #[snafu(display(
+        "Failed to cast default value {:?} to {}, reason: {}",
+        default_value,
+        target_type,
+        reason
+    ))]
+    CastDefaultValue {
+        default_value: Option<ColumnDefaultConstraint>,
+        target_type: ConcreteDataType,
+        reason: String,
+        source: datatypes::Error,
+        #[snafu(implicit)]
+        location: Location,
+    },
```

Comment:
We already have a reason, we can format the context in the reason string.
```suggestion
    #[snafu(display("{}", reason))]
    CastDefaultValue {
        reason: String,
        source: datatypes::Error,
        #[snafu(implicit)]
        location: Location,
    },
```